libcudacxx/docs/extended_api/synchronization_primitives/barrier/memcpy_async_tx.md

```diff
@@ -0,0 +1,87 @@
+---
+grand_parent: Extended API
+parent: Barriers
+---
+
+# `cuda::device::memcpy_async_tx`
+
+Defined in header `<cuda/barrier>`:
+
+```cuda
+template <typename T, size_t Alignment>
+inline __device__
+async_contract_fulfillment
+cuda::device::memcpy_async_tx(
+  T* dest,
+  const T* src,
+  cuda::aligned_size_t<Alignment> size,
+  cuda::barrier<cuda::thread_scope_block>& bar);
+```
+
+Copies `size` bytes from global memory `src` to shared memory `dest` and arrives
+on a shared memory barrier `bar`, updating its transaction count by `size`
+bytes.
+ 
```

Comment:
Docs should mention preconditions and effects.

Reply:
I did the preconditions.  I really don't grok what should be documented under effects, so I have left that as is. The current memcpy_async docs didn't have anything I could use as inspiration.